### ISSUE TITLE
1.5.0: Is API for EEx.Engine changed since 1.4?

### ISSUE THREAD

**Issue by grych** [elixir-lang/elixir#6391]
Is it an expected behaviour of `EEx.Engine` that it includes an `init` value in the middle expressions? Before 1.5.0, `init/1` value were added only at the beginning of the buffer, in the middle it was always empty string.

Observe the difference between 1.4.5 and 1.5.0 below:

#### Test engine
```
defmodule ExtendedEngine do
  use EEx.Engine

  def init(_) do
    "init:"
  end

  def handle_body(quoted) do
    quote do 
      unquote(quoted) <> ":end"
    end
  end
end
```

#### test.eex:
`<%= for x <- ["A", "B"] do %><%= x %><% end %>
`
#### 1.4.5:
```
Interactive Elixir (1.4.5) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> EEx.eval_file "text.eex", [], engine: ExtendedEngine
"init:AB\n:end"
```

#### 1.5.0:
```
Interactive Elixir (1.5.0) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> EEx.eval_file "text.eex", [], engine: ExtendedEngine
"init:init:Ainit:B\n:end"
```

If it is expected, shouldn't be `handle_body/1` called at the end of each middle expression? That would be consistent with `init/1`.

But I hope it is a bug, because previous behaviour was more intuitive.

### Environment

* Elixir & Erlang versions (elixir --version): Erlang/OTP 20, Elixir 1.4.5 and 1.5.0
* Operating system: macOS Sierra

The change was introduced here: https://github.com/elixir-lang/elixir/commit/b620445bb505a6da3235392647a09d0f0cb44b65

**Comment by josevalim** [comment]
Thank you for the report. @grych can you please give branch v1.5 (or master) a try? I want to make sure it will work as you expect.

**Comment by grych** [comment]
Hi @josevalim,
Perfect! 
```
Interactive Elixir (1.5.0) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)>  EEx.eval_file "text.eex", [], engine: ExtendedEngine
"init:AB\n:end"
```
And it fixed my issue grych/drab#31. Thanks!
